pwndbg/commands/spray.py

```diff
@@ -52,7 +59,28 @@ def spray(addr, length, value) -> None:
     else:
         value_bytes = cyclic(length, n=pwndbg.gdblib.arch.ptrsize)
 
-    try:
-        pwndbg.gdblib.memory.write(addr, value_bytes)
-    except gdb.MemoryError as e:
-        print(M.error(e))
+    if only_funcptrs:
+        try:
+            mem = pwndbg.gdblib.memory.read(addr, length)
+        except gdb.MemoryError as e:
+            print(M.error(e))
+            return
+
+        addresses_written = 0
+        ptrsize = pwndbg.gdblib.arch.ptrsize
+        for i in range(0, len(mem) - length % ptrsize, ptrsize):
```

Comment:
nit: I'd recommend parenthesis here just for clarity: `len(mem) - (length % ptrsize)`

Reply:
Fixed in https://github.com/pwndbg/pwndbg/pull/1819